src/Invictus.Testing.ResourceApi/ResourceApiAuthentication.cs

```diff
@@ -0,0 +1,137 @@
+﻿using GuardNet;
+using System.Net.Http;
+using System.Collections.Generic;
+using System.Globalization;
+using Newtonsoft.Json.Linq;
+using System.Threading.Tasks;
+using System;
+
+using ISecretProvider = Arcus.Security.Core.ISecretProvider;
+using Microsoft.IdentityModel.Clients.ActiveDirectory;
+
+namespace Codit.Testing.ResourceApi
+{
+    /// <summary>
+    /// Authentication representation to authenticate with resources running on Azure.
+    /// </summary>
+    public class ResourceApiAuthentication
```

Comment:
Can't we refactor our existing authentication and use that?

Reply:
Yes the token output of the Logic Apps is the same.

Reply:
Yes that is correct, but I think we should be able to re-use the authentication since it's the same, no?

Reply:
Yes a refactor of the Authenticator to give it generic naming and decouple from being Logic App specific would allow this PR to be re-factored and retested. Will proceed on that basis.